### PR TITLE
Add hx.isNumber

### LINKED ISSUE #240: Util: add hx.isNumber
For the sake of completeness, parallel with hx.isString etc.

Something like

```
hx.isNumber  = (x) => typeof x === 'number' || x instanceof Number
hx.isNumber(NaN) // => true
```

### PR DESCRIPTION
**Pull request by jmsmyth**
Resolves #240 (a request to add hx.isNumber)

### CODE REVIEW

File: modules/util/main/index.coffee [hunk 1]
```diff
@@ -141,6 +141,8 @@ hx.isObject = (obj) -> typeof obj is 'object' and not hx.isArray(obj) and obj is
 
 hx.isBoolean = (x) -> x is true or x is false or typeof x is 'boolean'
 
+hx.isNumber = (x) -> typeof x is 'number' || x instanceof Number
```
Comment (c-frater) [inline modules/util/main/index.coffee]: as it's coffee, should this not be 'or' instead of '||'? :)

File: modules/util/test/spec.coffee [hunk 2]
```diff
@@ -303,6 +303,23 @@ describe "Util", ->
     hx.isPlainObject(null).should.equal(false)
     hx.isPlainObject(undefined).should.equal(false)
 
+  it 'isNumber: should work', ->
+    class A
+      constructor: ->
+
+    hx.isNumber(new A).should.equal(false)
```
Comment (c-frater) [inline modules/util/test/spec.coffee]: Could also do `hx.isNumber(A)`? Although I guess this is probably equivalent to `hx.isNumber(->)`

Comment (jmsmyth) [inline modules/util/test/spec.coffee]: Yep, would be the same as `->`

**Comment by c-frater** [comment]
LGTM

**Comment by c-frater** [comment]
I've documented these changes:
https://github.com/ocadotechnology/hexagonjs-site/pull/100

**Pull request merged**